src/Magento/FunctionalTestingFramework/Test/Util/AnnotationExtractor.php

```diff
@@ -112,6 +117,30 @@ public function addStoryTitleToMap($annotations, $filename)
         }
     }
 
+    /**
+     * Appends TestCaseId or [NO TESTCASEID] to test titles (to prevent Allure collision).
+     * @param array  $annotations
+     * @param string $filename
+     * @return void
+     */
+    private function addTestCaseIdToTitle(&$annotations, $filename)
+    {
+        if (!isset($annotations['title'])) {
+            return;
+        }
+
+        $testCaseId = "[NO TESTCASEID]";
+
+        if (isset($annotations['testCaseId'])) {
+            $testCaseId = $annotations['testCaseId'][0];
+        }
+
+        $newTitle = "{$testCaseId} - " . $annotations['title'][0];
```

Comment:
This is total optional. would it be better to use ":" instead of "-" to concatenate test case ID and title, because there is a "-" inside test case ID and our commit message is formatted like "MAGETWO-123456: ".

Reply:
Good point, I will do this.